src/main/java/io/phasetwo/service/auth/storage/datastore/PhaseTwoExportImportManager.java

```diff
@@ -0,0 +1,263 @@
+                    } else {
+                        throw new ModelException("No user with email: " + invitation.getEmail());
+                    }
+                });
+    }
+
+    private void addMembers(RealmModel newRealm, OrganizationRepresentation organizationRepresentation, OrganizationModel org) {
+        organizationRepresentation.getMembers()
+                .forEach(member -> {
+                    var userModel = session.users().getUserByUsername(newRealm, member.getUsername());
+                    if (Objects.nonNull(userModel)) {
+                        org.grantMembership(userModel);
+                        member.getRoles()
+                                .stream()
+                                .map(org::getRoleByName)
+                                .forEach(organizationRoleModel -> organizationRoleModel.grantRole(userModel));
+                    } else {
+                        throw new ModelException("No user with username: " + member.getUsername());
+                    }
+                });
+    }
+
+    private static void createOrganizationIdp(RealmModel newRealm, LinkIdp idpLink, OrganizationModel org) {
+        if (Objects.nonNull(idpLink)) {
+            IdentityProviderModel idp = newRealm.getIdentityProviderByAlias(idpLink.getAlias());
+            if (!Strings.isNullOrEmpty(idpLink.getSyncMode())) {
+                idp.getConfig().put("syncMode", idpLink.getSyncMode());
+            }
+            if (!Strings.isNullOrEmpty(idpLink.getPostBrokerFlow())) {
+                String flowAlias = idpLink.getPostBrokerFlow();
+                if (flowAlias.trim().isEmpty()) {
+                    idp.setPostBrokerLoginFlowId(null);
+                } else {
+                    AuthenticationFlowModel flowModel = newRealm.getFlowByAlias(flowAlias);
+                    if (flowModel == null) {
+                        throw new ModelException("No available authentication flow with alias: " + flowAlias);
+                    }
+                    idp.setPostBrokerLoginFlowId(flowModel.getId());
+                }
+            }
+            idp.getConfig().put("hideOnLoginPage", "true");
+            idp.getConfig().put(ORG_OWNER_CONFIG_KEY, org.getId());
+
+            newRealm.updateIdentityProvider(idp);
+        }
+    }
+
+    private static void createOrganizationRoles(List<OrganizationRoleRepresentation> roles, OrganizationModel org) {
+        roles
+                .stream()
+                .filter(organizationRole ->
+                        Arrays.stream(OrganizationAdminAuth.DEFAULT_ORG_ROLES).noneMatch(role -> role.equals(organizationRole.getName())))
+                .forEach(organizationRole ->
+                {
+                    var role = org.addRole(organizationRole.getName());
+                    role.setDescription(organizationRole.getDescription());
+                });
+    }
+
+    protected AdminAuth authenticateRealmAdminRequest(HttpHeaders headers) {
```

Comment:
Where is this from? Was it `private` so we couldn't reuse it?

Reply:
It is imported from the original Keycloak code. Do you want to move it to a helper class?